static/src/javascripts/projects/common/modules/commercial/article-aside-adverts.js

```diff
@@ -3,66 +3,66 @@ define([
     'common/utils/$',
     'common/utils/$css',
     'common/utils/config',
-    'common/utils/fastdom-idle',
+    'common/utils/fastdom-promise',
     'common/modules/commercial/create-ad-slot',
-    'common/modules/commercial/commercial-features',
-    'lodash/objects/defaults'
+    'common/modules/commercial/commercial-features'
 ], function (
     Promise,
     $,
     $css,
     config,
-    idleFastdom,
+    fastdom,
     createAdSlot,
-    commercialFeatures,
-    defaults
+    commercialFeatures
 ) {
-    function init(options) {
-        var $mainCol, adType,
-            opts = defaults(
-                options || {},
-                {
-                    columnSelector: '.js-secondary-column',
-                    adSlotContainerSelector: '.js-ad-slot-container'
-                }
-            ),
-            $col        = $(opts.columnSelector),
-            colIsHidden = $col.length && $css($col, 'display') === 'none',
-            $componentsContainer,
-            $adSlotContainer;
+    var MIN_ARTICLE_HEIGHT = 1300;
+    var MIN_FOOTBALL_ARTICLE_HEIGHT = 2200;
+    var MIN_IMMERSIVE_ARTICLE_HEIGHT = 600;
+
+    var mainColumnSelector = '.js-content-main-column';
+    var rhColumnSelector = '.js-secondary-column';
+    var adSlotContainerSelector = '.js-ad-slot-container';
+    var componentsContainerSelector = '.js-components-container';
+
+    function init() {
+        var $col        = $(rhColumnSelector);
+        var colIsHidden = $col.length && $css($col, 'display') === 'none';
+        var $mainCol, $componentsContainer, $adSlotContainer;
 
         // are article aside ads disabled, or secondary column hidden?
         if (!commercialFeatures.articleAsideAdverts || colIsHidden) {
             return false;
         }
 
-        $mainCol = config.page.contentType === 'Article' ? $('.js-content-main-column') : false;
-        $componentsContainer = $('.js-components-container', '.js-secondary-column');
-        $adSlotContainer = $(opts.adSlotContainerSelector);
+        $mainCol = $(mainColumnSelector);
+        $componentsContainer = $(componentsContainerSelector, $col[0]);
```

Comment:
Actually the guard just above (`colIsHidden`) will prevent `$col[0]` from failing. This is arguably not very elegant.